depmapomics/mutations.py

```diff
@@ -157,17 +187,87 @@ def managingDuplicates(samples, failed, datatype, tracker):
     return renaming
 
 
+def aggregateMAFs(
+    wm, sampleset="all", mafcol=MAF_COL, keep_cols=MUTCOL_DEPMAP,
+):
+    """aggregate MAF files from terra
+
+    Args:
+        refworkspace (str): the reference workspace
+        sampleset (str, optional): the sample set to use. Defaults to 'all'.
+        mutCol (str, optional): the MAF column name. Defaults to "somatic_maf".
+        keep_cols (list, optional): which columns to keep in the aggregate MAF file. Defaults to MUTCOL_DEPMAP
+        
+    Returns:
+        aggregated_maf (df.DataFrame): aggregated MAF
+    """
+    print("aggregating MAF files")
+    sample_table = wm.get_samples()
+    samples_in_set = wm.get_sample_sets().loc[sampleset]["samples"]
+    sample_table = sample_table[sample_table.index.isin(samples_in_set)]
+    sample_table_valid = sample_table[~sample_table[mafcol].isna()]
+    na_samples = set(sample_table.index) - set(sample_table_valid.index)
+    print(str(len(na_samples)) + " samples don't have corresponding maf: ", na_samples)
+    all_mafs = []
+    le = len(sample_table_valid)
+    counter = 0
+    for name, row in sample_table_valid.iterrows():
+        # prints out progress bar
+        h.showcount(counter, le)
+        counter += 1
+        maf = pd.read_csv(row[mafcol])
+        maf[SAMPLEID] = name
+        if len(set(keep_cols) - set(maf.columns)) > 0:
```

Comment:
no need for column check. that should be a QC that can be run but not part of the regular pipeline